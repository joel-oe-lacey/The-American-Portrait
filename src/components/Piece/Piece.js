import React from 'react';
import './Piece.scss';

const Piece = ({ 
    primaryimageurl, 
    title, 
    objectid, 
    description, 
    century,
    dated,
    culture,
    classification,
    technique,
    medium,
    colors,
    creditline,
    labeltext
    }) => {

    return (
        <section key={objectid} className="piece-cont">
            <img alt={description} src={primaryimageurl} className="piece-img"/>
            <article className="piece-desc">
                <header className="desc-header">
                    <h1>{title}</h1>
                    <h3>{century}</h3>
                </header>
                <main className="desc-main">
                    <p>{labeltext ? labeltext : description}</p>
                </main>
                <footer className="desc-footer">
                    <h4>{`Dated: ${dated}`}</h4>
                    <h4>{`Medium: ${medium}`}</h4>
                    <h4>{`Technique: ${technique}`}</h4>
                    <h4>{creditline}</h4>
                </footer>
            </article>
        </section>
    )
}

export default Piece;