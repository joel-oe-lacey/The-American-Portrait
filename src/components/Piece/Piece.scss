.piece-cont {
    height: 100%;
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
    background-color: #202020;
    color: white;
}

.piece-img {
    height: 80%;
    width: 40%;
    object-fit: contain;
}

.piece-desc {
    height: 80%;
    width: 40%;
    border: 1px solid white;
}

.return-cont {
    height: 100%;
    width: 15%;
    display: flex;
    justify-content: flex-end;
    border-bottom: 1px solid white;
}

.return-btn {
    height: 25%;
    width: 100%;
    font-size: 1.5rem;
    text-decoration: none;
    color: white;
    background: transparent;
    text-align: center;
    padding: 20%;
    border-left: 1px solid white;
    border-bottom: 1px solid white;
}

.piece-grouping {
    height: 50%;
    width: 100%;
    display: flex;
    flex-direction: row;
}

.desc-header {
    height: 100%;
    width: 85%;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    border-bottom: 1px solid white;
    padding: 4%;

    h1 {
        font-weight: bold;
        font-size: 2.2vh;
    }
}

.desc-main {
    height: 25%;
    width: 100%;
    padding: 4%;
    line-height: 1.5;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
}

.desc-footer {
    height: 25%;
    width: 100%;
    width: 100%;
    padding: 4%;
    border-top: 1px solid white;
    display: flex;
    flex-flow: column;
    justify-content: space-between;
    font-size: .8em;

    h4 {
        width: 50%;
    }

    h3 {
        width: 100%;
    }
}