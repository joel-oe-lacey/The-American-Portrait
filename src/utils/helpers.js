export class ArtPiece {
    primaryimageurl;
    title;
    objectid;
    description;
    century;
    dated;
    culture;
    classification;
    technique;
    medium;
    colors;
    creditline;
    labeltext;

    constructor(rawArtData) {
        Object.assign(this, rawArtData);
    }
};

export const restructureArtPiece = (rawArtData) => {
    const {
        title,
        titles,
        objectid,
        description,
        provenance,
        commentary,
        labeltext,
        classification,
        creditline,
        century,
        culture,
        medium,
        videos,
        datebegin,
        dateend,
        dated,
        period,
        technique,
        colors,
        primaryimageurl,
        images,
    } = rawArtData;

    return {
        title,
        titles,
        objectid,
        description,
        provenance,
        commentary,
        labeltext,
        classification,
        creditline,
        century,
        culture,
        medium,
        videos,
        datebegin,
        dateend,
        dated,
        period,
        technique,
        colors,
        primaryimageurl,
        images
    }
}

export const bucketArtByDate = (allArt) => {
    const recompileCollection = (bucketedCollection) => {
        return Object.keys(bucketedCollection).reduce((flatColl, dateKey) => {
            flatColl = flatColl.concat(bucketedCollection[dateKey])
            return flatColl;
        }, [])
    }

    const bucketedCollection = allArt.reduce((bucketedArt, piece) => {
        const date = piece.dateend;
        switch(true) {
            case (date < 1800): 
                if (bucketedArt.toEighteenHundreds.length < 10) {
                    bucketedArt.toEighteenHundreds.push(piece)
                }
                break;
            case (date <= 1850):
                if (bucketedArt.eighteenFiftys.length < 10) {
                    bucketedArt.eighteenFiftys.push(piece)
                }
                break;
            case (date <= 1880):
                if (bucketedArt.eighteenEighty.length < 10) {
                    bucketedArt.eighteenEighty.push(piece)
                }
                break;
            case (date <= 1900):
                if (bucketedArt.toNineteenHundreds.length < 10) {
                    bucketedArt.toNineteenHundreds.push(piece)
                }
                break;
            case (date <= 1920):
                if (bucketedArt.toNineteenTwenties.length < 10) {
                    bucketedArt.toNineteenTwenties.push(piece)
                }
                break;
            case (date <= 1940):
                if (bucketedArt.toNineteenFourties.length < 10) {
                    bucketedArt.toNineteenFourties.push(piece)
                }
                break;
            case (date <= 1960):
                if (bucketedArt.toNineteenSixties.length < 10) {
                    bucketedArt.toNineteenSixties.push(piece)
                }
                break;
            case (date <= 1980):
                if (bucketedArt.toNineteenEighties.length < 10) {
                    bucketedArt.toNineteenEighties.push(piece)
                }
                break;
            case (date <= 2000):
                if (bucketedArt.toTwoThousands.length < 10) {
                    bucketedArt.toTwoThousands.push(piece)
                }
                break;
            case (date <= 2020):
                if (bucketedArt.toTwentyTwenties.length < 10) {
                    bucketedArt.toTwentyTwenties.push(piece)
                }
                break;
            default:
                break;
        }
        return bucketedArt;
    }, {
        "toEighteenHundreds": [],
        "toEighteenFiftys": [],
        "toEighteenEighty": [],
        "toNineteenHundreds": [],
        "toNineteenTwenties": [],
        "toNineteenFourties": [],
        "toNineteenSixties": [],
        "toNineteenEighties": [],
        "toTwoThousands": [],
        "toTwentyTwenties": [],
    })

    return recompileCollection(bucketedCollection);
}

