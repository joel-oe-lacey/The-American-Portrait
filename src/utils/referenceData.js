export const introText = ['People are trapped in history, and history is trapped in them" said James Baldwin in his 1953 essay "Stranger in the Village". At the time he spoke of the indivisibility of black culture from the larger collective American ethos and culture.', 
'A continuation of that allegory, I believe, also speaks to the importance of remaining in touch with our collective history. It, in similar duality, is a part of us all.', 'The American Portrait looks to reunite us with a visual piece of that history. Utilizing the Harvard Museum API, you can view a timeline of historic photos by state, ten per decade where available.','While incomplete and historically biased, these photos offer us at very least a partial window into some of the lives, events, and adolescent places that would go on to form the nation as we know it. ']

export const stateNames = ['Alabama', 'Alaska', 'Arizona', 'Arkansas', 'California', 'Colorado', 'Connecticut', 'Delaware', 'Florida', 'Georgia', 'Hawaii', 'Idaho', 'Illinois', 'Indiana', 'Iowa', 'Kansas', 'Kentucky', 'Louisiana', 'Maine', 'Maryland', 'Massachusetts', 'Michigan', 'Minnesota', 'Mississippi', 'Missouri', 'Montana', 'Nebraska', 'Nevada', 'New Hampshire', 'New Jersey', 'New Mexico', 'New York', 'North Carolina', 'North Dakota', 'Ohio', 'Oklahoma', 'Oregon', 'Pennsylvania', 'Puerto Rico', 'Rhode Island', 'South Carolina', 'South Dakota', 'Tennessee', 'Texas', 'Utah', 'Vermont', 'Virginia', 'Washington', 'West Virginia', 'Wisconsin', 'Wyoming']

export const mockCollectionPrimary = {
    "title": "House and [blue] Car, Near Akron, Alabama",
    "objectid": 354191,
    "creditline": "Harvard Art Museums/Fogg Museum, Fund for the Acquisition of Photographs",
    "century": "20th century",
    "technique": "Dye transfer print",
    "dateend": 1978,
    "dated": "1978, printed 1991",
    "people": [
        {
        "alphasort": "Christenberry, William",
        "birthplace": "Tascaloosa, Alabama",
        "name": "William Christenberry",
        "prefix": null,
        "personid": 62113,
        "gender": "unknown",
        "role": "Artist",
        "displayorder": 1,
        "culture": "American",
        "displaydate": "born 1936",
        "deathplace": null,
        "displayname": "William Christenberry"
        }
    ],
    "primaryimageurl": "https://nrs.harvard.edu/urn-3:HUAM:761794"
}

export const mockCollectionSecondary = {
    "title": "Night Spot, Marion, Alabama",
    "objectid": 354192,
    "creditline": "Harvard Art Museums/Fogg Museum, Fund for the Acquisition of Photographs",
    "century": "20th century",
    "technique": "Dye transfer print",
    "dateend": 1972,
    "dated": "1972, printed 1991",
    "people": [
        {
            "alphasort": "Christenberry, William",
            "birthplace": "Tascaloosa, Alabama",
            "name": "William Christenberry",
            "prefix": null,
            "personid": 62113,
            "gender": "unknown",
            "role": "Artist",
            "displayorder": 1,
            "culture": "American",
            "displaydate": "born 1936",
            "deathplace": null,
            "displayname": "William Christenberry"
        }
    ],
    "primaryimageurl": "https://nrs.harvard.edu/urn-3:HUAM:761793"
}