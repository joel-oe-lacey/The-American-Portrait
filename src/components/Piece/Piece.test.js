import React from 'react';
import { Piece } from './Piece';
import { shallow } from 'enzyme';
import { mockCollectionPrimary } from '../../utils/referenceData';

describe('Piece', () => {
    it('should match a snapshot as expected', () => {
        const wrapper = shallow(<Piece {...mockCollectionPrimary}/>)

        expect(wrapper).toMatchSnapshot()
    })
})
