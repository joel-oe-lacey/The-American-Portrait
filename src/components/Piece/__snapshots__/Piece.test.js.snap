// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Piece should match a snapshot as expected 1`] = `
<section
  className="piece-cont"
  key="354191"
>
  <img
    alt="House and [blue] Car, Near Akron, Alabama"
    className="piece-img"
    src="https://nrs.harvard.edu/urn-3:HUAM:761794"
  />
  <article
    className="piece-desc"
  >
    <span
      className="piece-grouping"
    >
      <header
        className="desc-header"
      >
        <h1>
          House and [blue] Car, Near Akron, Alabama
        </h1>
        <h3>
          Dated: 1978, printed 1991
        </h3>
      </header>
      <section
        className="return-cont"
      >
        <Link
          className="return-btn"
          to="/timeline"
        >
          X
        </Link>
      </section>
    </span>
    <main
      className="desc-main"
    >
      <h4>
        Artist: William Christenberry
      </h4>
      <h4>
        born 1936
      </h4>
      <h4>
        Born: Tascaloosa, Alabama
      </h4>
      <h4>
        Died: null
      </h4>
    </main>
    <footer
      className="desc-footer"
    >
      <h4>
        20th century
      </h4>
      <h4>
        Technique: Dye transfer print
      </h4>
      <h3>
        Harvard Art Museums/Fogg Museum, Fund for the Acquisition of Photographs
      </h3>
    </footer>
  </article>
</section>
`;
